oauth/resource_service_principal_federation_policy.go

```diff
@@ -0,0 +1,147 @@
+package oauth
+
+import (
+	"context"
+	"github.com/databricks/databricks-sdk-go/service/oauth2"
+	"github.com/databricks/terraform-provider-databricks/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"log"
+	"strings"
+)
+
+func ResourceServicePrincipalFederationPolicy() common.Resource {
```

Comment:
we really need to use plugin framework for all new resources: https://github.com/databricks/terraform-provider-databricks/blob/main/CONTRIBUTING.md#adding-a-new-resource